trailingstoploss_tp.py

```diff
@@ -108,39 +110,67 @@ def upgrade_config(thelogger, cfg):
                 cfg.write(cfgfile)
 
             thelogger.info("Upgraded section %s to have config list" % cfgsection)
-
+        if cfgsection.startswith("tsl_tp_") and cfg.has_option(cfgsection, "config"):
+            jsonconfiglist = list()
+            for configsectionconfig in json.loads(cfg.get(cfgsection, "config")):
+                if "sl-timeout" not in configsectionconfig:
+                    cfgsectionconfig = {
+                        "activation-percentage": configsectionconfig.get("activation-percentage"),
+                        "initial-stoploss-percentage": configsectionconfig.get("initial-stoploss-percentage"),
+                        "sl-timeout": "0",
+                        "sl-increment-factor": configsectionconfig.get("sl-increment-factor"),
+                        "tp-increment-factor": configsectionconfig.get("tp-increment-factor"),
+                    }
+                    jsonconfiglist.append(cfgsectionconfig)
+
+            if len(jsonconfiglist) > 0:
+                cfg.set(cfgsection, "config", json.dumps(jsonconfiglist))
+
+                with open(f"{datadir}/{program}.ini", "w+") as cfgfile:
+                    cfg.write(cfgfile)
+
+                thelogger.info("Upgraded section %s to have config list" % cfgsection)
```

Comment:
Netjes gedaan! Zou deze melding wel veranderen naar iets als 'Updates section %s to add sl-timeout'.